DiffMatchPatch.zig

```diff
@@ -2,9 +2,34 @@ const DiffMatchPatch = @This();
 
 const std = @import("std");
 const testing = std.testing;
+const Allocator = std.mem.Allocator;
 const ArrayListUnmanaged = std.ArrayListUnmanaged;
 const DiffList = ArrayListUnmanaged(Diff);
 
+/// Deinit an `ArrayListUnmanaged(Diff)` and the allocated slices of
+/// text in each `Diff`.
+pub fn deinitDiffList(allocator: Allocator, diffs: *DiffList) void {
+    defer diffs.deinit(allocator);
+    for (diffs.items) |d| {
+        if (d.text.len > 0) {
+            allocator.free(d.text);
+        }
```

Comment:
The free function is a no-op if the length is zero.
```suggestion
        allocator.free(d.text);
```